pkg/runtime/wfengine/activity.go

```diff
@@ -196,8 +207,14 @@ loop:
 				<-t.C
 			}
 			if completed {
+				elapsed := diag.ElapsedSince(start)
+				// activity completed, record count and latency metrics
+				diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, diag.ComponentName, diag.Activity, diag.StatusSuccess, elapsed)
 				break loop
 			} else {
+				elapsed := diag.ElapsedSince(start)
+				// Activity execution failed with recoverable error, record metrics
+				diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, diag.ComponentName, diag.Activity, diag.StatusRetryable, elapsed)
```

Comment:
For my understanding: How do we conclude here that it is for `retry` and not for `fail`?

Reply:
because we are returning a `recoverable error` which means Dapr will retry the activity execution.